packages/account-abstraction/src/BaseAccountAPI.ts

```diff
@@ -2,13 +2,15 @@ import { ethers, BigNumber, BigNumberish } from 'ethers'
 import { Provider } from '@ethersproject/providers'
 import { UserOperation } from '@biconomy/core-types' // review
 import { ClientConfig } from './ClientConfig' // ClientConfig is needed in this design
+import { BytesLike } from "@ethersproject/bytes";
+import { EntryPoint } from '@account-abstraction/contracts'
 
 import {
   EntryPointContractV100,
   SmartWalletFactoryV100,
   SmartWalletContractV100
 } from '@biconomy/ethers-lib'
-import { TransactionDetailsForUserOp } from './TransactionDetailsForUserOp'
+import { TransactionDetailsForBatchUserOp } from './TransactionDetailsForUserOp'
```

Comment:
we will always send a batch, array of targets values datas and in case of single transaction we will send single element arrays.